### DISCUSSION TITLE
Lesson 7: Error when installing smartcontractkit/chainlink-brownie-contracts

### DISCUSSION THREAD

**Discussion by c0llinx** [Cyfrin/foundry-full-course-cu#964] [Q&A] [answered]
I'm trying to install the smartcontractkit/chainlink-brownie-contracts using the command below but I keep getting an error

`forge install smartcontractkit/chainlink-brownie-contracts@0.7.1 --no-commit`

Here's the full error:

`forge install smartcontractkit/chainlink-brownie-contracts@0.7.1 --no-commit`
`Installing chainlink-brownie-contracts in /foundry-f23/foundry-fund-me/lib/chainlink-brownie-contracts (url: Some("https://github.com/smartcontractkit/chainlink-brownie-contracts"), tag: Some("0.7.1"))
fatal: please make sure that the .gitmodules file is in the working tree
Error: 
git submodule exited with code 128: Reactivating local git directory for submodule 'lib/chainlink-brownie-contracts'`

Pls, I need some assistance

**Comment by w3rc** [comment]
Can you remove the version and just use `forge install smartcontractkit/chainlink-brownie-contracts --no-commit`?

**Reply by c0llinx** [reply]
Tried it but got the same error.

**Comment by 0xusmanf** [comment]
1. Check if the `.gitmodules` file exists in your working directory. If it doesn't, you'll need to restore it. You can do this by running the command `git restore .gitmodules`

1. If the `.gitmodules` file is present but you're still encountering the error, it's possible that the submodule name or path specified in the `.gitmodules` file doesn't match the actual name or path of the submodule. Make sure that the submodule name and path match exactly

### Example `.gitmodules` file
```gitmodules
[submodule "lib/forge-std"]
	path = lib/forge-std
	url = https://github.com/foundry-rs/forge-std
	branch = v1.5.3
```

**Reply by c0llinx** [reply]
Done that.

**Reply by 0xusmanf** [reply]
Another workaround is to create a new project by `forge init` and copy-paste your existing `.sol` files into the new project and try again.

**Reply by tobezhanabi** [reply]
Ive tried this, still got the git submodele error

**Comment by hosseinmirzapur** [comment]
> I'm trying to install the smartcontractkit/chainlink-brownie-contracts using the command below but I keep getting an error
> 
> `forge install smartcontractkit/chainlink-brownie-contracts@0.7.1 --no-commit`
> 
> Here's the full error:
> 
> `forge install smartcontractkit/chainlink-brownie-contracts@0.7.1 --no-commit` `Installing chainlink-brownie-contracts in /foundry-f23/foundry-fund-me/lib/chainlink-brownie-contracts (url: Some("https://github.com/smartcontractkit/chainlink-brownie-contracts"), tag: Some("0.7.1")) fatal: please make sure that the .gitmodules file is in the working tree Error: git submodule exited with code 128: Reactivating local git directory for submodule 'lib/chainlink-brownie-contracts'`
> 
> Pls, I need some assistance

I had the same error. Save each and every file your coding in, then remove the cache folder then run the command. worked for me, i hope it works for you as well

**Comment by Ehsan-The-Coder** [comment, marked as answer]
### **i was facing the same issue but resolve with one simple way** 

```
forge install smartcontractkit/chainlink-brownie-contracts@0.8.0 --no-commit
Installing chainlink-brownie-contracts in D:/BlockChain/Project/Showcase/FoundryFullCourse/FundMe/lib\chainlink-brownie-contracts (url: Some("https://github.com/smartcontractkit/chainlink-brownie-contracts"), tag: Some("0.8.0"))
Cloning into 'D:/BlockChain/Project/Showcase/FoundryFullCourse/FundMe/lib/chainlink-brownie-contracts'...
remote: Enumerating objects: 32923, done.
remote: Counting objects: 100% (2524/2524), done.
remote: Compressing objects: 100% (1050/1050), done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 3081 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
fatal: clone of 'https://github.com/smartcontractkit/chainlink-brownie-contracts' into submodule path 'D:/BlockChain/Project/Showcase/FoundryFullCourse/FundMe/lib/chainlink-brownie-contracts' failed
Error: 
git submodule exited with code 128
```



**Issue:** my issue was that i initiated git in the FoundryFullCourse inside this i have all lesson folders

**Solution:** initiate the git with `git init` in the working directory and then try to install with 
``` 
forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit
```

**Reply by Equious** [reply]
MVP. This just helped me out with another course entirely lol

**Reply by Nikzadkhan** [reply]
just copy paths from the url like this 
<img width="359" alt="zeplien test" src="https://github.com/user-attachments/assets/67e4596a-dbac-489f-804c-398546402746">

**Reply by EngrPips** [reply]
Interesting.

**Comment by Fishyuhao** [comment]
reboot terminal

**Comment by Javadyakuza** [comment]
just commit the changes and then run the command without `--no-commit` flag.

**Reply by owanemi** [reply]
this worked for me

**Comment by oladsdan** [comment]
i am still having this issue please how do i resolve

**Reply by EngrPips** [reply]
It is strange to me how people are stating how they resolved this. For me, it has always been an internet connection problem, and I just retry whenever I know there is less traffic, and boom, it will just work.

**Comment by oladsdan** [comment]
"git config --global http.postBuffer 524288000" that was the code that solved mine

**Reply by EngrPips** [reply]
Interesting. This makes sense to me.

**Reply by deepmindd** [reply]
huh, that worked for me. thanks

**Reply by EngrPips** [reply]
Awesome

**Comment by legendarycode3** [comment]
𝐢 𝐰𝐚𝐬 𝐟𝐚𝐜𝐢𝐧𝐠 𝐭𝐡𝐢𝐬 𝐬𝐚𝐦𝐞 𝐢𝐬𝐬𝐮𝐞 𝐛𝐞𝐟𝐨𝐫𝐞, 𝐰𝐡𝐞𝐧 𝐭𝐫𝐲𝐢𝐧𝐠 𝐭𝐨 𝐰𝐨𝐫𝐤 𝐰𝐢𝐭𝐡 𝐅𝐨𝐮𝐧𝐝𝐫𝐲 𝐨𝐧 𝐦𝐲 𝐖𝐢𝐧𝐝𝐨𝐰𝐬 𝐏𝐜 (𝐫𝐚𝐭𝐡𝐞𝐫 𝐭𝐡𝐧 𝐦𝐲 𝐦𝐚𝐜 𝐩𝐜)

Here is the error that pops up below :
<img width="707" height="301" alt="image" src="https://github.com/user-attachments/assets/03de7d01-242d-4e3d-837c-12318a1a86db" />
The reason why it doesnot install is basically when your path is too long that the lib file can not be located when tying to install the chainlink-brownie-contracts file.  
Here is the way to do it rather after you have navigated to the folder and file of your project, on  Git Bash terminal, make sure you reduced the path  you trying to locate . Below is the example
<img width="445" height="160" alt="image" src="https://github.com/user-attachments/assets/afcf8e70-2c24-47db-8c26-14d823769610" />

**Reply by EngrPips** [reply]
Thanks for this heads-up.
